### ISSUE TITLE
[ES|QL] Improve `show functions` command with some additional columns

### ISSUE THREAD

**Issue by dej611** [elastic/elasticsearch#100621]
### Description

After #99507 the output of `show functions` became very useful.
While testing with the Kibana code I've noticed a couple more columns that would be nice to have:
* `minParams: number` - this is useful to describe variadic functions who still require few parameters to be defined
  * Some examples here are `concat(...)` with `minParams: 1`, `case` with `minParams: 3` or `cidr_match` with `minParams: 2`
* `examples`: some short example on how to use the function
* `aggregation`: true/false can be useful to identify where it can be used (`stats` vs `eval/row/where`) 
* `supportsWildcard`: true/false for each argument

**Comment by bpintea** [comment]
Functionality dropped in https://github.com/elastic/elasticsearch/pull/113967.
